src/Magick.NET/Formats/Png/PngWriteDefines.cs

```diff
@@ -0,0 +1,122 @@
+    /// For compression level 0 (quality value less than 10), the Huffman-only strategy is used, which is the fastest but not necessarily the worst compression.
+    /// </summary>
+    public uint? CompressionLevel { get; set; }
+
+    /// <summary>
+    /// Gets or sets the compression strategy for the PNG image.
+    /// </summary>
+    public PngCompressionStrategy? CompressionStrategy { get; set; }
+
+    /// <summary>
+    /// Gets or sets the chunks to be excluded.
+    /// </summary>
+    public PngChunkFlags? ExcludeChunks { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the PNG decoder should ignore the CRC when writing the image.
+    /// </summary>
+    public bool IgnoreCrc { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the iCCP chunk should be preserved when writing the image.
+    /// </summary>
+    public bool PreserveiCCP { get; set; }
+
+    /// <summary>
+    ///  Gets or sets a value indicating whether ColorMap should be preserve when writing the image.
+    /// </summary>
+    public bool PreserveColorMap { get; set; }
+
+    /// <summary>
+    /// Gets the format where the defines are for.
+    /// </summary>
+    public MagickFormat Format
+        => MagickFormat.Png;
+
+    /// <summary>
+    /// Gets the defines that should be set as a define on an image.
+    /// </summary>
+    public IEnumerable<IDefine> Defines
+    {
+        get
+        {
+            if (BitDepth.HasValue)
+                yield return new MagickDefine(Format, "bit-depth", BitDepth.Value);
+
+            if (ColorType.HasValue)
+            {
+                var colorTypeValue = GetPngColorTypeValue(ColorType.Value);
+                yield return new MagickDefine(Format, "color-type", (int)colorTypeValue);
+            }
+
+            if (CompressionFilter.HasValue)
+                yield return new MagickDefine(Format, "compression-filter", (int)CompressionFilter.Value);
+
+            if (CompressionLevel.HasValue)
+                yield return new MagickDefine(Format, "compression-level", CompressionLevel.Value);
+
+            if (CompressionStrategy.HasValue)
+                yield return new MagickDefine(Format, "compression-strategy", (int)CompressionStrategy.Value);
+
+            if (ExcludeChunks.HasValue)
+                yield return new MagickDefine(Format, "exclude-chunks", EnumHelper.ConvertFlags(ExcludeChunks.Value));
+
+            if (IgnoreCrc)
+                yield return new MagickDefine(Format, "ignore-crc", IgnoreCrc);
+
+            if (PreserveiCCP)
+                yield return new MagickDefine(Format, "preserve-iCCP", PreserveiCCP);
+
+            if (PreserveColorMap)
+                yield return new MagickDefine(Format, "preserve-colormap", PreserveColorMap);
+        }
+    }
+
+    private PngColorType GetPngColorTypeValue(ColorType colorType)
```

Comment:
I think we can remove the enumeration and directly return an integer instead? We don't need the `PngColorType` enum than. This is a much better solution then what I was thinking we should do.

p.s. You could change this to `=> colorType switch` and get rid of `{` and `}`.

Reply:
Hello, this week has been very busy for me, and I haven't had time to make the final adjustments to the PR (pull request), but I am committed to finishing it this week.

Regarding the use of the PngColorType enum, as a best practice, I always avoid using numbers or strings directly in the code. I prefer to use enums or constants to improve readability and maintenance. However, I understand that in framework-level code and libraries, performance is crucial. Therefore, I will remove the PngColorType enum. I believe I will be able to finish this PR this week.

Thank you for your attention.

Reply:
Take all the time you need.